vectordb_bench/backend/clients/memorydb/memorydb.py

```diff
@@ -0,0 +1,254 @@
+                client = client.get_primaries()[0].redis_connection
+
+            if kwargs.get("replicas", False):
+                # Return client and host name for each replica
+                return [(c.redis_connection, c.host) for c in replicas]
+
+        else:
+            client = Redis(
+                host=self.db_config["host"],
+                port=self.db_config["port"],
+                db=0,
+                ssl=self.db_config["ssl"],
+                password=self.db_config["password"],
+                ssl_ca_certs=self.db_config["ssl_ca_certs"],
+                ssl_cert_reqs=None,
+            )
+            client.execute_command("PING")
+        return client
+
+    @contextmanager
+    def init(self) -> Generator[None, None, None]:
+        """ create and destory connections to database.
+
+        Examples:
+            >>> with self.init():
+            >>>     self.insert_embeddings()
+        """
+        self.conn = self.get_client()
+        search_param = self.case_config.search_param()
+        if search_param["ef_runtime"]:
+            self.ef_runtime_str = f'EF_RUNTIME {search_param["ef_runtime"]}'
+        else:
+            self.ef_runtime_str = ""
+        yield
+        self.conn.close()
+        self.conn = None
+
+    def ready_to_load(self) -> bool:
+        pass
+
+    def optimize(self) -> None:
+        self._post_insert()
+
+    def insert_embeddings(
+        self,
+        embeddings: list[list[float]],
+        metadata: list[int],
+        **kwargs: Any,
+    ) -> Tuple[int, Optional[Exception]]:
+        """Insert embeddings into the database.
+        Should call self.init() first.
+        """
+
+        try:
+            with self.conn.pipeline(transaction=False) as pipe:
+                for i, embedding in enumerate(embeddings):
+                    embedding = np.array(embedding).astype(np.float32)
+                    pipe.hset(metadata[i], mapping = {
+                        "id": str(metadata[i]),
+                        "metadata": metadata[i], 
+                        "vector": embedding.tobytes(),
+                    })
+                    # Execute the pipe so we don't keep too much in memory at once
+                    if (i + 1) % self.insert_batch_size == 0:
+                        pipe.execute()
+
+                pipe.execute()
+                result_len = i + 1
+        except Exception as e:
+            return 0, e
+        
+        return result_len, None
+    
+    def _post_insert(self):
+        """Wait for indexing to finish"""
+        client = self.get_client(primary=True)
+        log.info("Waiting for background indexing to finish")
+        args = (self.wait_for_no_activity, 5, "", client)
+        self.wait_until(*args)
+        if not self.db_config["cmd"]:
+            replica_clients = self.get_client(replicas=True)
+            for rc, host_name in replica_clients:
+                args = (self.wait_for_no_activity, 5, "", rc)
```

Comment:
Here it's `5`, but on https://github.com/zilliztech/VectorDBBench/pull/339/files#diff-1638e273a6d7eee7b4412bf79bda9112da86d53e18d1b650b8b80a2e1df4f7e6R59 it's 3, unsure why or the impact of the different values.

Reply:
Its just that `wait_for_no_activity` takes longer than the `flushall` operation. We can keep the same value for both.